packages/web3-core/src/web3_config.ts

```diff
@@ -255,20 +322,64 @@ export abstract class Web3Config
 		this._config.defaultChain = val;
 	}
 
+	/**
+	 * The default hardfork property is used for signing transactions locally.
+	 *
+	 * Will return the default hardfork
+	 *
+	 * @returns {string}
+	 *
+	 * The default hardfork property can be one of the following:
+	 * - `chainstart`
+	 * - `homestead`
+	 * - `dao`
+	 * - `tangerineWhistle`
+	 * - `spuriousDragon`
+	 * - `byzantium`
+	 * - `constantinople`
+	 * - `petersburg`
+	 * - `istanbul`
+	 * - `berlin`
+	 * - `london`
+	 *
+	 * Default is `london`
+	 */
 	public get defaultHardfork() {
 		return this._config.defaultHardfork;
 	}
 
+	/**
+	 *
+	 * @param {string} val
+	 */
 	public set defaultHardfork(val) {
 		this._triggerConfigChange('defaultHardfork', val);
 
 		this._config.defaultHardfork = val;
 	}
 
+	/**
+	 *
+	 * Will get the default common property
+	 *
+	 *The default common property does contain the following Common object:
+	 *
+	 * `customChain` - `Object`: The custom chain properties
+	 * - `name` - `string`: (optional) The name of the chain
+	 * - `networkId` - `number`: Network ID of the custom chain
+	 * - `chainId` - `number`: Chain ID of the custom chain
+	 * - `baseChain` - `string`: (optional) mainnet, goerli, kovan, rinkeby, or ropsten
+	 * - `hardfork` - `string`: (optional) chainstart, homestead, dao, tangerineWhistle, spuriousDragon, byzantium, constantinople, petersburg, istanbul, berlin, or london
+	 */
 	public get defaultCommon() {
 		return this._config.defaultCommon;
 	}
 
+	/**
+	 * Will set the default common property
+	 *
+	 * @returns {Record<string,unknown}
```

Comment:
Same here don't need to document the return type, will be done from the Typescript. 

```suggestion
```